topics/statistics/tutorials/loris_model/tutorial.md

```diff
@@ -0,0 +1,273 @@
+
+However, there are limits to how predictive age might be, particularly for extreme values. For example, patients over a certain age may share similar health characteristics, and further increases in age may not significantly add predictive value. Truncating age to a maximum value (like 85) helps avoid overemphasizing small differences between very old patients, where the added predictive power might be negligible.
+
+## Response
+The Response feature is a categorical target variable indicating whether patients benefited from immune checkpoint blockade (ICB) therapy, classified as 0 (no benefit) or 1 (benefit). The model is trained using all previous features to predict patient outcomes when ICB is chosen as the treatment.
+
+# Prepare environment and get the data 
+> <comment-title>Preprocessing the raw data</comment-title>
+>
+> The raw data published by ({% cite Chang2024 %}) can be found here: 
+> [LORIS raw dataset](https://github.com/rootchang/LORIS/blob/main/02.Input/AllData.xlsx)
+> 
+> We preprocessed the raw data using a Python script to:
+> 1) Extract the `Chowell_train` and `Chowell_test` tab from the excel file.
+> 2) Select the 7 features (`TMB`, `Systemic Therapy History`, `Albumin`, `Cancer Type`, `NLR`, `Age`, and `Response`) important for building the model.
+> 3) Truncate the values for `Age`, `NLR`, and `TMB`.
+> 4) Encode `Cancer Type` using one-hot encoding. 
+> 5) Save the dataset as a .tsv file.
+>
+> A jupyter notebook for preprocessing can be found at Dockstore: [LORIS_preprocessing](https://dockstore.org/notebooks/github.com/paulocilasjr/pycaret-use-case/preprocessing:main?tab=info)
+>
+{:  .comment}
+
+> <hands-on-title> Environment and Data Upload </hands-on-title>
+>
+> 1. Create a new history for this tutorial. If you are not inspired, you can name it *LORIS model classifier*.
+>
+>    {% snippet faqs/galaxy/histories_create_new.md %}
+>
+> 2. Import the files from Zenodo or from the shared data library
+>
+>    ```
+>    https://zenodo.org/records/13885908/files/Chowell_train_Response.tsv
+>    https://zenodo.org/records/13885908/files/Chowell_test_Response.tsv
+>    ```
+>
+>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
+>
+> 3. Check that the data format assigned for the file is **tsv**.
+>    If it is not, follow the Changing the datatype tip.
+>
+>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="datatypes" %}
+>
+> 5. Add a tag (`LORIS model dataset`) to the dataset corresponding to `Chowell_train_Response.tsv` and `Chowell_test_Response.tsv`
+>    This is important to trace back on what dataset the model was built on.
+>
+>    {% snippet faqs/galaxy/datasets_add_tag.md %}
+>
+{: .hands_on}
+
+# Using PyCaret Model Comparison Tool
+> <hands-on-title> Task description </hands-on-title>
+>
+> 1. {% tool [PyCaret Model Comparison](.bx.psu.edu/repos/paulo_lyra_jr/pycaret_model_comparison/PyCaret_Model_Comparison/2024.3.3.2+0) %} with the following parameters:
```

Comment:
the tool-id looks broken

Reply:
![Screenshot 2024-12-02 at 4 06 49 PM](https://github.com/user-attachments/assets/326e8488-5b0e-4882-8d00-353782db08a2)
This one is odd. I changed because of lint test. See here the change.

Reply:
What was the old linting error?

Reply:
This link is def wrong, can you please change it back?

Reply:
I looked other tutorial to follow same pattern. Found this one: [regression_machinelearning](https://raw.githubusercontent.com/paulocilasjr/training-material/refs/heads/main/topics/statistics/tutorials/regression_machinelearning/tutorial.md).

it will satisfy the test and wont be broken.